examples/chip-tool/commands/common/Commands.cpp

```diff
@@ -176,14 +192,14 @@ CHIP_ERROR Commands::RunCommand(NodeId localId, NodeId remoteId, int argc, char
         // set the variable to true, which will cause it to block indefinitely.
         //
         command->UpdateWaitForResponse(true);
+#if CONFIG_USE_SEPARATE_EVENTLOOP
         chip::DeviceLayer::PlatformMgr().ScheduleWork(RunQueuedCommand, reinterpret_cast<intptr_t>(command));
         command->WaitForResponse(command->GetWaitDurationInSeconds());
-        err = command->GetCommandExitStatus();
-        if (err != CHIP_NO_ERROR)
-        {
-            ChipLogError(chipTool, "Run command failure: %s", chip::ErrorStr(err));
-            ExitNow();
-        }
+#else  // CONFIG_USE_SEPARATE_EVENTLOOP
+        err = command->Run();
+        SuccessOrExit(err);
+        command->ScheduleWaitForResponse(command->GetWaitDurationInSeconds());
```

Comment:
This set of lines highlight how much a mirror image of the other these two approaches are, and capture their essence:

Either you asynchronously schedule work and synchronously wait for the response (separate event loop), or synchronously do work and asynchronously wait for the response (same event loop).

Reply:
> This set of lines highlight how much a mirror image of the other these two approaches are, and capture their essence:
> 
> Either you asynchronously schedule work and synchronously wait for the response (separate event loop), or synchronously do work and asynchronously wait for the response (same event loop).

Could you also add this as a comment in the code (maybe Command.h at the first `#if`)?